src/redux/actions/loginAction.js

```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+import { FETCH_USERS_REQUEST, FETCH_USERS_SUCCESS, FETCH_USERS_FAILURE, CLOSE_MESSAGE } from '../types/loginTypes';
+
+export const loginAction = (data) => {
+	return async (dispatch) => {
+		dispatch(fetchUsersRequest());
+		await axios
+			.post(`https://octopus-bn-backend.herokuapp.com/api/v1/auth/signin`, data)
+			.then((response) => {
+				// response.data is the user
+				const user = response.data;
+				localStorage.setItem('token', response.data.token);
+				dispatch(fetchUsersSuccess(user));
+			})
+			.catch((error) => {
+				// error.message is the error message
+				dispatch(fetchUsersFailure(error.response.data.error));
+			});
+	};
+};
+
+export const fetchUsersRequest = () => {
+	return {
+		type: FETCH_USERS_REQUEST
+	};
+};
+
+export const fetchUsersSuccess = (user) => {
+	return {
+		type: FETCH_USERS_SUCCESS,
+		payload: user
+	};
+};
+
+export const fetchUsersFailure = (error) => {
```

Comment:
Please do tweak actions creatores to make a sense.
```suggestion
export const loginFailure = (error) => {
```